### PR TITLE
Move pinning to the back of the kaon and enhance example

### PR DESCRIPTION
**Pull request by felipesere**
I thought we needed a more explicit example about pattern matching and pinning.

@ukutaht @Maikon @iamvery @gemcfadyen @RabeaGleissner

### CODE REVIEW

File: lib/koans/10_pattern_matching.ex [hunk 1]
```diff
@@ -117,4 +99,39 @@ defmodule PatternMatching do
...
+      (^pinned_variable) -> "The number One"
+      (2) -> "The number Two"
+      (number) -> "The number #{number}"
+    end
+    assert example.(1) == ___
+    assert example.(2) == ___
+    assert example.(3) == ___
+  end
+
+  koan "Pinning works on anywhere one would match, including 'case'" do
```
Comment (iamvery) [inline lib/koans/10_pattern_matching.ex]: Either remove the word "on" or reword to "work on anything"

File: lib/koans/10_pattern_matching.ex [hunk 2]
```diff
@@ -117,4 +99,39 @@ defmodule PatternMatching do
...
+    pinned_variable = 1
+    other_variable = 1
+    result = case other_variable do
```
Comment (iamvery) [inline lib/koans/10_pattern_matching.ex]: I sort of feel like the `other_variable` is distracting here.

File: lib/koans/10_pattern_matching.ex [hunk 3]
```diff
@@ -7,24 +7,6 @@ defmodule PatternMatching do
...
-
-  koan "A variable can be rebound" do
-    a = 1
-    a = 2
-    assert a == ___
-  end
-
-  koan "A variable can be pinned to prevent it from being rebounded" do
-    a = 1
-    assert_raise MatchError, fn() ->
```
Comment (iamvery) [inline lib/koans/10_pattern_matching.ex]: I do still sort of like this to illustrate what happens when a pinned match fails. Maybe introduce this before some of the more advanced examples?

File: lib/koans/10_pattern_matching.ex (hunk 1 above)
Comment (felipesere) [inline lib/koans/10_pattern_matching.ex]: One of these days I'll actually check my spelling before pushing...

File: lib/koans/10_pattern_matching.ex (hunk 2 above)
Comment (felipesere) [inline lib/koans/10_pattern_matching.ex]: Should I just do 'case 1 do'?

Comment (iamvery) [inline lib/koans/10_pattern_matching.ex]: Ya, but that's a bit meh, huh?

Comment (iamvery) [inline lib/koans/10_pattern_matching.ex]: Ya, but that's a bit meh, huh?

File: lib/koans/10_pattern_matching.ex (hunk 3 above)
Comment (felipesere) [inline lib/koans/10_pattern_matching.ex]: Done, but at the bottom of the koan. The idea is to introduce how to use it (well) and then what happens when it fails.

Comment (iamvery) [inline lib/koans/10_pattern_matching.ex]: Sounds great! 🎉

**Pull request merged**